### DISCUSSION TITLE
Emoji pasting on X11

### DISCUSSION THREAD

**Discussion by swiftgeek** [dail8859/NotepadNext#338] [Q&A] [answered]
Not sure if reproducible on other machines yet, or master even, but on my linux/x11 machine with NotepadNext 0.6 pasting this freezes/hangs notepadnext window
```
🪲
https://a.b.c/
```

Can anyone reproduce?

**Comment by dail8859** [comment, marked as answer]
Reproducible on Windows as well.

**Reply by dail8859** [reply]
Finally got around to this. The bug will be fixed in the next release. Since the 🪲 character requires 4 bytes, the indexes into the file got messed up when trying to highlight the URL, causing the code to get into an infinite loop past the end of the file. This fix should make regular expression searching much more stable when the file contains emojis and other unicode characters.

**Reply by swiftgeek** [reply]
Tested 9433150ce958a973cb4c28a0579e2977963189c6 and it seems to be working for me, thanks for squashing this 🪲.

**Reply by dail8859** [reply]
@swiftgeek Thanks for confirming!
